jvm/diffuser/src/main/java/com/spotify/diffuser/diffuser/Diffuser.java

```diff
@@ -53,8 +54,7 @@ private Diffuser(DidChange<A> didChange, Effect<A> sideEffect) {
 
     this.effect =
         value -> {
-          A cachedValue = cache.get();
-          if (cachedValue == null || didChange.test(cachedValue, value)) {
```

Comment:
Is this null check not necessary?

Reply:
@togi @pettermahlen What do you'll think about this? ^

Reply:
It's been a long time, but I'm pretty sure that the original intention was that `null` means 'not set'/'first value', and that null values would be illegal. If so, this PR changes those semantics and might thus be considered to need a new major version if you're a bit purist. On the other hand, I'm not sure how clearly those semantics were expressed anywhere.